mysql/resource_grant.go

```diff
@@ -712,9 +759,19 @@ func hasConflictingGrants(ctx context.Context, db *sql.DB, desiredGrant MySQLGra
 				continue
 			}
 		}
-		return dbGrant, nil
+
+		// For some reason, MySQL separates privileges into multiple lines
+		// So to normalize them, we need to combine them into a single MySQLGrant
+		if result != nil {
+			result, err = combineGrants(result, dbGrant)
+			if err != nil {
+				return nil, err
```

Comment:
Please always consider wrapping an error, when you're returning it. That way, it's easier to trace the source by searching for (unique) strings.